pandas_path/accessor.py

```diff
@@ -5,112 +5,130 @@
 import pandas as pd
 
 
-@pd.api.extensions.register_series_accessor("path")
-@pd.api.extensions.register_index_accessor("path")
-class PathAccessor:
-    """Adds a `.path` accessor to Series and Index objects so that all of the methods
-        from pathlib.Path are available.
+def path_accessor_factory(path_class, *args, **kwargs):
+    class PathAccessor:
+        """Adds a `.path` accessor to Series and Index objects so that all of the methods
+            from pathlib.Path are available.
 
-        Register with pandas by importing. No other changes necessary.
+            Register with pandas by importing. No other changes necessary.
 
-        Note: Will return a `str` rather than a Path object so that joining with existing
-            dataframes, etc. will continue to work.
+            Note: Will return a `str` rather than a Path object so that joining with existing
+                dataframes, etc. will continue to work.
 
-        Note: May not be performant on extremely long lists since we cast to `Path`
-        for every operation.
+            Note: May not be performant on extremely long lists since we cast to `Path`
+            for every operation.
 
-    Raises
-    ------
-    AttributeError
-        If the method we try to use on the accessor is not a function or property or
-        is not an attribute on Path, raise an AttributeError.
-    """
+        Raises
+        ------
+        AttributeError
+            If the method we try to use on the accessor is not a function or property or
+            is not an attribute on Path, raise an AttributeError.
+        """
```

Comment:
Do the class name or docstring matter? (i.e., do they show up somewhere for users to look at?) If so, could potentially dynamically assign `__name__` and `__doc__` in the `register_path_accessor` function.

Reply:
Nice call, here's the IPyhton `?` results for the `win` version now:

```
Type:        PureWindowsPathAccessor
String form: <pandas_path.accessor.path_accessor_factory.<locals>.PathAccessor object at 0x7fe7a8fdccd0>
File:        ~/code/pandas-path/pandas_path/accessor.py
Docstring:  
A `.win` accessor on Series and Index objects so that all of the methods
    from PureWindowsPath are available.

    Note: Always will return a `str` rather than a PureWindowsPath object so that joining with existing
        dataframes, etc. will continue to work.

    Note: May not be performant on extremely long lists since we cast to `PureWindowsPath`
    for every operation.

Raises
------
AttributeError
    If the method we try to use on the accessor is not a function or property or
    is not an attribute on PureWindowsPath, raise an AttributeError.
```